pkg/handlers/send_odp_event.go

```diff
@@ -36,26 +36,26 @@ func SendOdpEvent(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	body, err := getResponseBody(r)
+	body, err := getRequestOdpEvent(r)
 	if err != nil {
 		RenderError(err, http.StatusBadRequest, w, r)
 		return
 	}
 
 	success := optlyClient.SendOdpEvent(body.Action, body.Type, body.Identifiers, body.Data)
-	if err != nil {
+	if !success {
 		RenderError(err, http.StatusInternalServerError, w, r)
```

Comment:
This "err" should be fixed later when go-sdk returns proper error for `SendOdpEvent`. Let's keep it for now.